client/state/posts/likes/reducer.js

```diff
@@ -38,6 +45,113 @@ export const items = createReducer(
 				},
 			};
 		},
+		[ POST_LIKE ]: ( state, { siteId, postId } ) => {
```

Comment:
do you think it would  simplify this reducer to use two nested keyedReducers? Here would be example of handling `POST_LIKE`:

```js
export const items = keyedReducer( 'siteId', keyedReducer( 'postId', createReducer(
  { likes: undefined, iLike: false, found: 0 }, // default state
  {
    [  POST_LIKES_RECEIVE ]: (state, { likes, iLike, found } ) => ({
        likes: ...
        iLike,
        found,
    }),
    [ POST_LIKE ]: state => 
      state.iLike 
         ? state
         : { ...state, iLike: true, found: state.found + 1  };
  }
})))
```

Reply:
FFTI  this whole thread of ideas here if you don't like em, but heres another related idea:

if we extracted the `createReducer`, then we could test that directly and the test data would be less verbose.

e.g.

```js
// reducer.js file
export const postLikes = createReducer(
  { likes: undefined, iLike: false, found: 0 }, // default state
  {
    [  POST_LIKES_RECEIVE ]: (state, { likes, iLike, found } ) => ({
        likes: ...
        iLike,
        found,
    }),
  } 
);

export const items = keyedReducer( 'siteId', keyedReducer( 'postId' ), postLikes );

// test/reducer.js
// test data does not need to be nested anymore with { 1: { 1 : {} } } 
```